client/adventureMap/CInfoBar.h

```diff
@@ -66,14 +69,14 @@ class CInfoBar : public CIntObject
 
 	class VisibleHeroInfo : public CVisibleInfo
 	{
-		std::shared_ptr<CHeroTooltip> heroTooltip;
+		std::variant<std::shared_ptr<CHeroTooltip>, std::shared_ptr<CInteractableHeroTooltip>> heroTooltip;
```

Comment:
Just use `shared_ptr<CIntObject>` instead of variant

Reply:
OK, though I think variant reflects the intent better